tests/integration/kubernetes/k8s-confidential-attestation.bats

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bats
+# Copyright 2024 IBM Corporation
+# Copyright 2024 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/../../common.bash"
+load "${BATS_TEST_DIRNAME}/lib.sh"
+load "${BATS_TEST_DIRNAME}/confidential_common.sh"
+load "${BATS_TEST_DIRNAME}/confidential_kbs.sh"
+load "${BATS_TEST_DIRNAME}/tests_common.sh"
+
+export KBS="${KBS:-false}"
+export test_key="aatest"
+export SUPPORTED_TEE_HYPERVISORS=("qemu-sev" "qemu-snp" "qemu-tdx" "qemu-se")
+export SUPPORTED_NON_TEE_HYPERVISORS=("qemu")
+export KATA_HYPERVISOR="${KATA_HYPERVISOR:-qemu}"
+export AA_KBC="${AA_KBC:-cc_kbc}"
+
+setup() {
+	confidential_setup || skip "Test not supported for ${KATA_HYPERVISOR}."
+
+	if [ "${KBS}" = "false" ]; then
+		skip "Test skipped as KBS not setup"
+	fi
+
+	setup_common
+	get_pod_config_dir
+
+#	setup_unencrypted_confidential_pod
+
+	export K8S_TEST_YAML="${pod_config_dir}/pod-attestable.yaml"
+	# Schedule on a known node so that later it can print the system's logs for
+	# debugging.
+	set_node "$K8S_TEST_YAML" "$node"
+
+	if [[ " ${SUPPORTED_NON_TEE_HYPERVISORS} " =~ " ${KATA_HYPERVISOR} " ]]; then
+		info "Need to apply image annotations"
+		export add_non_tee_annotation="io.katacontainers.config.hypervisor.image"
+		export non_tee_image_path="/opt/kata/share/kata-containers/kata-containers-confidential.img"
+		yq write -i "${K8S_TEST_YAML}" "metadata.annotations[${add_non_tee_annotation}]" "${non_tee_image_path}"
```

Comment:
@GabyCT @stevenhorsman, I recommend to call `set_metadata_annotation` to apply image annotation. https://github.com/kata-containers/kata-containers/blob/c163d9f1145b07123cdd77b967cd01784fef0c3f/tests/integration/kubernetes/lib.sh#L177-L193